test/config_params/test_cofig_params.py

```diff
@@ -0,0 +1,368 @@
+    logging.getLogger().info("Splunk received %s events in the last minute" +
+                             " with u_id=%s",
+                             len(events), u_id)
+    assert len(events) == expected
+
+
+@pytest.mark.parametrize("test_input,expected", [
+    (None, 1),
+    ("test_source_type", 1)
+])
+def test_splunk_source_type(setup, test_input, expected):
+    '''
+    Test that docker logs can be indexed with the specified
+    sourcetype successfully. If no source is specified, the default
+    "httpevent" is used
+    '''
+
+    logging.getLogger().info("testing test_splunk_source_type input={0} \
+                expected={1} event(s)".format(test_input, expected))
+    u_id = str(uuid.uuid4())
+
+    file_path = setup["fifo_path"]
+    start_log_producer_from_input(file_path, [("test source", False)], u_id)
+
+    options = {}
+    if test_input:
+        options = {"splunk-sourcetype": test_input}
+    request_start_logging(file_path,
+                          setup["splunk_hec_url"],
+                          setup["splunk_hec_token"],
+                          options=options)
+
+    sourcetype = test_input if test_input else "httpevent"
+
+    # wait for 10 seconds to allow messages to be sent
+    time.sleep(10)
+    request_stop_logging(file_path)
+
+    # check that events get to splunk
+    events = check_events_from_splunk(id="sourcetype={0} {1}"
+                                         .format(sourcetype, u_id),
+                                      start_time="-2m@m",
+                                      url=setup["splunkd_url"],
+                                      user=setup["splunk_user"],
+                                      password=setup["splunk_password"])
+    logging.getLogger().info("Splunk received %s events in the last minute" +
+                             " with u_id=%s",
+                             len(events), u_id)
+    assert len(events) == expected
+
+
+def test_splunk_ca(setup):
+    '''
+    Test that docker logging plugin can use the server certificate to
+    verify the server identity
+
+    The server cert used here is the default CA shipping in splunk
+    '''
+    logging.getLogger().info("testing test_splunk_ca")
+    u_id = str(uuid.uuid4())
+
+    file_path = setup["fifo_path"]
+    start_log_producer_from_input(file_path, [("test ca", False)], u_id)
+    current_dir = os.path.dirname(os.path.realpath(__file__))
+
+    options = {
+        "splunk-insecureskipverify": "false",
+        "splunk-capath": current_dir + "/cacert.pem",
+        "splunk-caname": "SplunkServerDefaultCert"
+    }
+
+    parsed_url = urlparse(setup["splunk_hec_url"])
+    hec_ip = parsed_url.hostname
+    hec_port = parsed_url.port
+    splunk_hec_url = "https://SplunkServerDefaultCert:{0}".format(hec_port)
+
+    if "SplunkServerDefaultCert" not in open('/etc/hosts').read():
+        file = open("/etc/hosts", "a")
+        file.write("{0}\tSplunkServerDefaultCert".format(hec_ip))
```

Comment:
Do you want to clean this up after the test? Or better, create a dup hosts file and back up the original and switch it back during tear down.

Reply:
I was thinking about this and decided not to do it because:
- Our tests should be run on a test machine so it really doesn't matter to add one entry to the host. I also don't see much damage from adding one entry to the host file.
- Create a dup file and back up from the original has more steps and it is difficult to make sure the process is atomic.

Reply:
Ok, I can live with that.